share/spack/qa/run-flake8

```diff
@@ -21,9 +21,12 @@ fi
 changed=$(git diff --name-only develop... | grep '.py$')
 
 # Exempt url lines in changed packages from overlong line errors.
+# Exempt functions defined with '@when' decorator from redefinition errors.
 for file in $changed; do
     if [[ $file = *package.py ]]; then
-        perl -i~ -pe 's/^(\s*url\s*=.*)$/\1  # NOQA: ignore=E501/' $file;
+        cp "$file" "$file~"
+        perl -i -pe 's/^(\s*url\s*=.*)$/\1  # NOQA: ignore=E501/' $file
```

Comment:
@adamjstewart: Including the tilde in multiple perl replacements causes successive backups to overwrite previous ones.  Doing the backup at the very beginning and performing a sequence of in-place replacements avoids this problem.